### DISCUSSION TITLE
[MU4 Issue] Keyboard shortcut consistency suggestion: alt+shift+arrow-key

### DISCUSSION THREAD

**Discussion by WillyWonksters** [musescore/MuseScore#11783] [General]
The left/right keys move from chord/note to chord/note. Shift+left/right performs the same movement but also selects those chords/notes.
Since Alt+arrow-key also moves up/down across the individual notes in a chord, shouldn’t alt+shift+arrow-key also select those individual notes?

**Platform information**
MuseScore 4.0.0 OS: Debian GNU/Linux 11 (bullseye), Arch.: x86_64, MuseScore version (64-bit): 4.0.0-2059995262, revision: github-musescore-musescore-cafefd7

**Comment by MarcSabatella** [comment]
There's something logical about this to be sure, but - the way Shift+Left/Right work, it's always about a range selection, and since each press of Left or Right by definition moves the cursor to a different time position, each press of Shift+Left/Right will always change an endpoint of the range.  That wouldn't be true for Alt+Shfit+Left/Right.  You might have a range selected and then Alt+Right moves the selection to a hairpin - what would that mean?  Or, you might be moving from note to note within a single chord which doesn't change the time position, so this would have no effect on the range endpoints.

But, one thing this does bring up is the idea that making a *list* selection - normally by Ctrl+clicking individual elements that may or may not be contiguous - is difficult or impossible to do by keyboard.  That's because any attempt to navigate to another element after selecting one loses the original selection, because selection *is* navigation.  Meaning, list selections are pretty much not accessible to blind users (except to the extent you can create them by right-click & the Select menu).

But, what if Alt+Shift+Left/Right (or some other shortcut) moved the "selection cursor" but left the existing selection alone otherwise?  So if you are currently on an element, then you press Alt+Shift+Right, it would leave that element selected and then also select the next element.  From there, Alt+Right could navigate while leaving that first element selected.  Then the next Alt+Shift+Right would add the current element to the list selection and move the cursor again, etc.  So Alt+Shfit+Right would basically be the equivalent of "Ctrl+click plus select next element as the cursor", Alt+right would always move the active element but leave anything else that was previously selected.

Sounds complicated, I know, but in my imagination, it feelsrelatively natural, and would be one solution to a problem I don't think anyone has really thought much about solving before.

**Reply by shoogle** [reply]
@MarcSabatella, I agree: we can't change range selections in the way the OP suggested but we should make list selections accessible via the keyboard. As you said, we need a shortcut that changes the __current item__ without changing the __selection state__. In File Explorer this is done using Ctrl and the arrow keys. You can then use Ctrl+Spacebar to select or deselect multiple items.

![image](https://user-images.githubusercontent.com/11011881/170754903-0a61d779-3cb0-47de-b327-62bd02d4026b.png)

- __Current item__ - the item that represents the navigation position (i.e. has keyboard focus)
    - That's the 'Users' folder in the screenshot. (Notice the thin blue outline.)
- __Selected item(s)__ - the item(s) that a subsequent action would be performed upon (e.g. modified or deleted).
    - In the screenshot, that's the three folders with the solid blue highlight.

Unfortunately we're already using Ctrl+arrow for other things in MuseScore, like changing the pitch by an octave or moving left or right by a full measure [1]. We can't use Ctrl+Alt+arrow because that's often used as a system shortcut to switch desktop or rotate the screen, so Alt+Shift+arrow might indeed be a good choice, although I was also considering using that for a different navigation command to go to the [Next Chronological Element](https://musescore.org/en/node/330850).

1. I think this shortcut was borrowed from text editors where Ctrl+arrow navigates by word or paragraph. The Score View in MuseScore is like a weird hybrid between an item view and a text editing region: sometimes you want one form of interaction, at other times you want the other form.

**Reply by WillyWonksters** [reply]
Ok great! list selection is exactly what I had in mind, I just didn't know the name. And yeah, the ctrl+spacebar solution would be great if only it didn't conflict with ctrl+arrow.

FYI to the curious: Ctrl+arrow also navigates by word in the browser and many other places too, while ctrl+shift+arrow selects text by word.

**Reply by Magnus26** [reply]
> FYI to the curious: Ctrl+arrow also navigates by word in the browser and many other places too, while ctrl+shift+arrow selects text by word.

Hello! Yes, Ctrl+arrow should move the cursor to the next or previous bar, or the staff above or below.

**Reply by Magnus26** [reply]
> Unfortunately we're already using Ctrl+arrow for other things in MuseScore, like changing the pitch by an octave or moving left or right by a full measure [1]. We can't use Ctrl+Alt+arrow because that's often used as a system shortcut to switch desktop or rotate the screen, so Alt+Shift+arrow might indeed be a good choice, although I was also considering using that for a different navigation command to go to the [Next Chronological Element](https://musescore.org/en/node/330850).
> 
>     1. I think this shortcut was borrowed from text editors where Ctrl+arrow navigates by word or paragraph. The Score View in MuseScore is like a weird hybrid between an item view and a text editing region: sometimes you want one form of interaction, at other times you want the other form.

Yes, it is this weird hybrid that I think MuseScore Studio should get away from and instead make it as close as possible to a word processor, a program type that often beginners are already used to. Some twenty years ago I suggested to the Sibelius development team that arrows alone should only be used to move the cursor and not move objects, and that Alt+arrow should be used for the latter, something I believe is now followed in Dorico. Therefore I think it is bad to have Ctrl+uparrow move a note. The modifier keys should as much as possible be connected to one or a few special meanings: Ctrl does except "control" often mean "much"; Alt does except "alternative" mean "move" (at least in Dorico), and Meta (Windows) often means "very much". Ctrl+Alt could then mean "move much", and I believe it is possible also in Windows to make Ctrl+Alt+arrow available for this, in that case with the program Power Toys.
